cumulusci/core/keychain/encrypted_file_project_keychain.py

```diff
@@ -423,36 +515,63 @@ def _rename_alias_in_default_service_file(
 
     def _load_services(self) -> None:
         """Load services (and migrate old ones if present)"""
+        self._load_services_from_environment()
+
         if not (self.global_config_dir / "services").is_dir():
             self._create_default_service_files()
             self._create_services_dir_structure()
             self._migrate_services()
+
         self._load_service_files()
 
-    def _set_service(self, service_type, alias, service_config):
+    def _set_service(
+        self, service_type, alias, service_config, save=True, config_encrypted=False
+    ):
+        if "services" not in self.config:
+            self.config["services"] = {}
```

Comment:
Should we do this once in `__init__`?

Reply:
It would probably be best.

Reply:
Looks like BaseProjectKeychain.__init__() is already doing that for us.